middle_end/flambda/compilenv_deps/reg_width_things.ml

```diff
@@ -602,11 +605,20 @@ module Simple = struct
   let export t = find_data t
 
   let import map (data : exported) =
-    let simple = map data.simple in
-    let data : Simple_data.t =
-      { simple; rec_info = data.rec_info; }
-    in
-    Table.add !grand_table_of_simples data
+    (* The grand table of [Simple]s only holds those with [Rec_info]. *)
+    let old_simple = data.simple in
+    (* [old_simple] never has [Rec_info].  See [Simple_data], above. *)
+    assert (Id.flags old_simple <> simple_flags);
+    let t = map old_simple in
+    let rec_info = data.rec_info in
+    if Rec_info.is_initial rec_info then begin
+      assert (Id.flags t <> simple_flags);
+      t
+    end else begin
+      assert (Id.flags t = simple_flags);
```

Comment:
I think this assertion can never be true. I'm quite sure that import maps will always preserve flags, so since `old_simple` doesn't have the simple flag then neither will `t`.
Am I missing something ?

Reply:
Well spotted, this was nonsense.